serverAkkaHttp/src/main/scala/RPCRouter.scala

```diff
@@ -16,7 +16,21 @@ import scala.concurrent.Future
 
 import io.circe.Json
 
-object RouteGenerators {
+trait Router extends RPCServer with PathMacro with MetaDataMacro {
+  def tp: Seq[String]
+  def methodsMetaData: Map[String, MethodMetaData]
+  def routes: autowire.Core.Router[Json]
+  def path: String = tp.last
+
+  def buildRoute: Route = handleExceptions(exceptionHandler) {
+    pathPrefix(path) {
+      methodsMetaData map {
```

Comment:
@calippo do you have any guarantee that `methodsMetaData` is non empty?

Otherwise when reducing on an empty `List` this will throw. `reduceOption` is a safer choice.